[Search] Extract connectors to package

sphilipse opened the pull request:
## Summary

This refactors connectors, enterprise search and serverless search so we no longer need to activate enterprise search for serverless search.

sphilipse committed:
[Search] Extract connectors to package

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

TattdCodeMonkey committed:
[Serverless Search] Getting Started: Separate cloud id panel (#165170)

## Summary

Moved the cloud id from the API key panel to its own cloud details panel
with the url as well.

### Screenshots

![image](https://github.com/elastic/kibana/assets/1972968/9db88c94-7285-4e7d-8a4b-106a8f1707f1)

![image](https://github.com/elastic/kibana/assets/1972968/20486279-3f49-44d7-bf9d-c63bb39dabe1)

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

TattdCodeMonkey reviewed (approved):
Some comments but overall LGTM, great work!

pzl committed:
[Fleet] Classify index not found as file not found too (#165469)

dmlemeshko committed:
test api/status for capacity (#164864)

## Summary

Adding API capacity test for `GET /api/status`


Stats:
```
"warmupAvgResponseTime": 39,
"rpsAtWarmup":10,
"warmupDuration":30,
"rpsMax":695,
"responseTimeMetric":"85%",
"threshold1ResponseTime":1000,
"rpsAtThreshold1":435,
"threshold2ResponseTime":3000,
"rpsAtThreshold2":475,
"threshold3ResponseTime":5000,
"rpsAtThreshold3":495
```

dgieselaar committed:
[Logs] use ISO timestamp for Assistant messages (#165475)

CoenWarmer committed:
[Observability Ai Assistant] Onboarding & Action Menu Improvements (#165459)

jpdjere committed:
[Security Solution] Fix Related integrations Cypress test (#165481)

jloleysens committed:
[Jest tests] Update create serverless root helpers to use `kill` (#165467)

## Summary

Follow up on https://github.com/elastic/kibana/pull/165316, address
https://github.com/elastic/kibana/pull/165316#discussion_r1312403704

* Use the `kill` option in Jest integration test utilities for
serverless roots
* Fix a typo in a type import
* Add the `waitForReady` flag to optionally wait until the cluster is
ready to server requests (default: `false`)

## TODO

- [x] Add a test for `waitForReady`

kobelb committed:
Allow kibana-tech-leads manage Serverless Release Pipeline (#165491)

christineweng committed:
[Security Solution] Expandable flyout - Update index field in analyzer preview (#165398)

## Summary

This PR addresses a bug that when a rule is created using data view,
analyzer preview is blank. This is due to a check on if index exists
before rendering the analyzer preview. This PR updated the index field
from `kibana.alert.rule.parameter.index` to 'kibana.alert.rule.indices`,
the later is introduced in https://github.com/elastic/kibana/pull/130929
and is available when a rule is created using either index patterns or
data view.

**How to reproduce the bug**
- Refer to bug report https://github.com/elastic/kibana/issues/164829

**How to test**
- Create a rule using data view
- Generate some alerts
- Go to alerts page, expand a row in alerts table
- Go to Visualization -> Analyzer preview, the analyzer preview tree
should be present


![image](https://github.com/elastic/kibana/assets/18648970/cbe0668e-335d-436a-992c-8970e75a3635)


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

PhilippeOberti committed:
[Security Solution] expandable flyout - add paywall to prevalence details (#165382)

kobelb committed:
Removing the double quotes from the build.filter (#165500)

@brianseeders figured this out. We tried out this change by modifying
the pipeline directly, and it triggered the pipeline:
https://buildkite.com/elastic/kibana-serverless-release

kobelb committed:
Skipping pre-command when running the Upload pipeline (#165505)

Context: https://elastic.slack.com/archives/C5UDAFZQU/p1693587045671319

Commit:
GitHub create-deploy-tag workflow: Show commit SHA (#165460)

kobelb committed:
Echoing out BUILDKITE_COMMAND (#165506)

joeypoon committed:
[Security Solution] fix endpoint heartbeat flag not correctly setting (#165419)

jbudz committed:
[ci] Stricter check on pre_command skip (#165519)

When buildkite-agent is uploading a pipeline we can skip setting up our
node environment. This stricter check avoids matching on similarly named
steps, e.g. our storybooks upload.

walterra committed:
[ML] Transforms: Improve transform list reloading behavior. (#164296)

## Summary

- If the transform list fails to load, it does no longer show a
non-refreshable full page error. Instead the error is shown in an inline
callout and the refresh button is still present.
- `AuthorizationProvider` is gone and has been replaced by a custom
hook `useTransformCapabilities`. All client side code no longer relies
on `privileges` being present but makes use of `capabilities` (like
`canGetTransform` etc.). The custom route to fetch privileges and
capabilities is also gone, instead capabilities are retrieved from
Kibana's own `application.capabilities.transform` which we register
server side.
- Refactors all remote data fetching to use `react-query`. This gets
rid of the custom code to refresh the transform list using observables,
instead all CRUD actions now make use of `react-query`'s `useMutation`
and trigger a cache invalidation of the transform list data to initiate
a refetch. The `useApi` hook is gone, instead we now have specific hooks
for data fetching that wrap `useQuery` (`useGetTransform`,
`useGetTransformStats` etc.) and the existing hooks for actions have
been refactored to use `useMutation` (`useStartTransforms`,
`useStopTransforms` etc.).
- Toasts for "success" messages have been removed.
- All tests that made use of `toMatchSnapshot` have been refactored
away from `enzyme` to `react-testing-lib` and no longer rely on
snapshots, instead we make basic assertions on the rendered components.

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine committed:
skip failing test suite (#165386)

kibanamachine committed:
skip failing test suite (#165396)

kibanamachine committed:
skip failing test suite (#165476)

dplumlee committed:
[Security Solution] Fixes MITRE page description (#165363)

kibanamachine committed:
[api-docs] 2023-09-02 Daily api_docs build (#165536)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/448

YulNaumenko committed:
Extracted DiscoverGrid to a package named @kbn/unified-data-table as UnifiedDataTable component (#163211)

Current PR includes the next set of changes:

1. Moved `DiscoverGrid` component to a package `@kbn/unified-data-table`
and added `@elastic/kibana-data-discovery` as code owners.
2. Changed `@kbn/unified-data-table` package naming for data grid
related components and methods to correspond `UnifiedDataTable` instead
of `Discover`.

3. Moved hooks `useColumns` and `useRowHeightsOptions` to a package as
its logic belongs to `UnifiedDataTable`.
4. Renamed `DiscoverGridContext` to `UnifiedDataTableContext`.
5. Extended `UnifiedDataTable` interface and functionality with the next
customization options:
- `renderDocumentView?: (displayedRows:
DataTableRecord[],displayedColumns: string[]) => JSX.Element |
undefined;` - callback to render DocumentView when the document is
expanded
- `configRowHeight?: number;` - optional value for providing
configuration setting for UnifiedDataTable rows height
- `showMultiFields?: boolean;` - optional value for providing
configuration setting for enabling to display the complex fields in the
table. Default is true.
- `maxDocFieldsDisplayed?: number;` - optional value for providing
configuration setting for maximum number of document fields to display
in the table. Default is 50.
- `externalControlColumns?: EuiDataGridControlColumn[];` - optional
value for providing EuiDataGridControlColumn list of the additional
leading control columns. UnifiedDataTable includes two control columns:
Open Details and Select.
<img width="522" alt="Screenshot 2023-08-22 at 2 26 57 PM"
src="https://github.com/elastic/kibana/assets/55110838/d796b9c8-2fef-4bcc-a3c9-9f5cc6349ab9">

- `externalAdditionalControls?: React.ReactNode;` - optional value for
providing the additional controls available in the UnifiedDataTable
toolbar to manage it's records or state. UnifiedDataTable includes
Columns, Sorting and Bulk Actions.
<img width="673" alt="Screenshot 2023-08-22 at 2 40 28 PM"
src="https://github.com/elastic/kibana/assets/55110838/f7ac0c87-5310-49dd-9084-1ce01ca0f366">

- `rowsPerPageOptions?: number[];` - optional list of number type values
to set custom UnifiedDataTable paging options to display the records per
page.
- `renderCustomGridBody?: (args: EuiDataGridCustomBodyProps) =>
React.ReactNode;` - An optional function called to completely customize
and control the rendering of EuiDataGrid's body and cell placement.
<img width="1658" alt="Screenshot 2023-08-22 at 2 50 27 PM"
src="https://github.com/elastic/kibana/assets/55110838/14adc18d-73af-40f5-9859-b3c708e265b1">

- `trailingControlColumns?: EuiDataGridControlColumn[];` - optional list
of the `EuiDataGridControlColumn` type for setting trailing control
columns standard for `EuiDataGrid`.
- `visibleCellActions?: number;` - optional value for a custom number of
the visible cell actions in the table
<img width="497" alt="Screenshot 2023-08-22 at 2 45 49 PM"
src="https://github.com/elastic/kibana/assets/55110838/57ef3ad9-7401-46bb-9b38-cc8cca2e6a24">

- `externalCustomRenderers?: Record<string,(props:
EuiDataGridCellValueElementProps) => React.ReactNode>;` - an optional
settings for a specified fields rendering like links. Applied only for
the listed fields rendering:
<img width="1121" alt="Screenshot 2023-08-22 at 2 51 07 PM"
src="https://github.com/elastic/kibana/assets/55110838/77501eae-3046-4a2c-90e1-2db487c21e2c">

- `consumer` - optional string value for the name of the
`UnifiedDataTable` consumer component or application.
6. Extended `UnifiedDataGrid` services with the two additional:
    `storage: Storage;`
    `data: DataPublicPluginStart; `
replaced `core: CoreStart;` with `theme: ThemeServiceStart;`, because
`core` is used only to get `theme`
7. Replaced `DocumentView` property with `renderDocumentView?:
(displayedRows: DataTableRecord[],displayedColumns: string[]) =>
JSX.Element | undefined;` callback function, which allows not to use
`DiscoverGridFlyout` definition for the documents rendering.
```
    /**
   * Document detail view component
   */
  DocumentView?: typeof DiscoverGridFlyout;
```
8. Removed the next properties from the data table interface, because it
was used to render DiscoverGridFlyout:
```
   /**
   * Filters applied by saved search embeddable
   */
  filters?: Filter[];
  /**
   * Query applied by KQL bar or text based editor
   */
  query?: Query | AggregateQuery;
  /**
   * Saved search id used for links to single doc and surrounding docs in the flyout
   */
  savedSearchId?: string;
```
9. Added usage examples and interface description to README file.
10. Changed grid styles naming from `.dscDiscoverGrid*` to
`.udtDataTable*`
11. Migrated discover plugin to use `UnifiedDataTable` instead of
`DiscoverGrid`

Extra changes were needed to avoid the circular dependancies:
- moved `DocViewFilterFn` and `FieldMapping` from discover plugin to
`packages/kbn-discover-utils/src/types.ts`
- added own `export type SortOrder = [string, string];` to avoid deps
for saved-search plugin

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
[api-docs] 2023-09-03 Daily api_docs build (#165544)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/449

CohenIdo committed:
[Cloud Security] fix index template for serverless

dgieselaar committed:
[Observability AI Assistant] Save/open Lens embeddable (#165542)

dgieselaar committed:
[Observability AI Assistant] Don't grant access to connectors (#165540)

mistic committed:
chore(NA): skip currently unstable serverless Cypress tests (#165424)

This is a simple PR skipping some Cypress tests that are currently
flaky.

kibanamachine committed:
[api-docs] 2023-09-04 Daily api_docs build (#165551)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/450

ashokaditya committed:
[Security Solution][Endpoint] Fix and unskip flaky test (#165466)

## Summary

Fixes and un-skips a test. Ensures that the test data doesn't create an
existing index. If it still does, as a temp. measure we're dismissing
the error toast that is blocking the button so the rest of the test can
continue.

closes elastic/kibana/issues/139260

**flakey test runner**

https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/3036
x 200 ( all green)

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

elasticmachine committed:
[main] Sync bundled packages with Package Storage (#165371)

Automated by
https://internal-ci.elastic.co/job/package_storage/job/sync-bundled-packages-job/job/main/6707/

Co-authored-by: apmmachine <infra-root+apmmachine@elastic.co>

js-jankisalvi committed:
[Cases] Add contentManagement dependency to Observability and Security solution plugins (#165225)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

ashokaditya committed:
[Security Solution][Endpoint][Response Actions]fix tray collapse on history page (#165260)

## Summary

Fixes a bug on the response actions history page where the details tray
does not collapse on a single click.

**before** 
elastic/kibana/issues/165058

**after**

![response-actions-hisotry-collapse](https://github.com/elastic/kibana/assets/1849116/71ee1663-c087-41ef-b7c0-3c9be65f7ef6)

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

elasticmachine committed:
[main] Sync bundled packages with Package Storage (#165562)

Automated by
https://internal-ci.elastic.co/job/package_storage/job/sync-bundled-packages-job/job/main/6737/

Co-authored-by: apmmachine <infra-root+apmmachine@elastic.co>

machadoum committed:
[Security Solutions] Add Maps, Visualization Library to Security Serverless IA (#165315)

## Summary

* Add Maps, Visualization Library to Security Serverless IA
* Add callout to dashboards page
* Add `dashboardsLandingCalloutComponent$` to the service contract

*Graphs link was removed from the scope of this PR because it is
disabled on serverless.*


<img width="520" alt="Screenshot 2023-08-31 at 13 33 34"
src="https://github.com/elastic/kibana/assets/1490444/b5e159dd-86e7-44dc-8a1c-ae96ca43fba8">
<img width="1191" alt="Screenshot 2023-08-31 at 13 33 46"
src="https://github.com/elastic/kibana/assets/1490444/b4c21192-d4fe-4dc2-af30-aedc8a9fede8">
<img width="1517" alt="Screenshot 2023-08-31 at 13 35 48"
src="https://github.com/elastic/kibana/assets/1490444/d0a43dc1-b0ae-475f-8350-50ee891363e3">
<img width="515" alt="Screenshot 2023-08-31 at 13 33 25"
src="https://github.com/elastic/kibana/assets/1490444/102bdfdc-01f1-434c-9946-342c9cbb8e0b">

### How to test it

#### Servereless
1. Open the dashboards page
2. Check if the callout is there
3. Check if all links work
4. Open the menu and check if the changes are reflected
5. Check if all links work
6. Open the project settings landing and check if the changes are
reflected
7. Check if all links work

#### ESS
1. Open the dashboards page
2. Nothing should have changed
3. Open menu
4. Nothing should have changed
5. Open manage landing page
6. Nothing should have changed

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)

dmlemeshko committed:
[ftr] update docs for serverless run on MKI (#165295)

## Summary

Adding information about `skipMKI` tag and how to run tests against MKI
projects from local machine

ashokaditya committed:
[Osquery] Use fleet actions client to access fleet indices in osquery (#164845)

## Summary

Replaces calls to fleet indices with `FleetActionsClient` that was
introduced in elastic/kibana/pull/158833

### Checklist
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Ashokaditya <ashokaditya.mohanty@elastic.co>

pheyos committed:
Serverless test readme - link to kbn-es readme (#165572)

## Summary

This PR updates the serverless test readme and adds a link to the
@kbn-es readme.

ppisljar committed:
changing defaults (#164957)

CoenWarmer committed:
Move function naming to American English (#165561)

kibanamachine committed:
skip failing test suite (#139879)

sphilipse committed:
Fix typing issues

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'
